src/utils/react-query-factory.js

```diff
@@ -0,0 +1,93 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { createQueryKeyStore } from '@lukemorales/query-key-factory';
+import { fetchCourseMetadata } from '../components/app/routes/queries/courseMetadata';
+import { fetchUserEntitlements } from '../components/app/routes/queries/userEntitlements';
+import { fetchCanRedeem } from '../components/app/routes/queries/canRedeemCourse';
+import { fetchEnterpriseLearnerData } from '../components/app/routes/queries/enterpriseLearner';
+import { fetchEnterpriseCourseEnrollments } from '../components/app/routes/queries/enterpriseCourseEnrollments';
+import { fetchBrowseAndRequestConfiguration } from '../components/app/routes/queries/subsidies/browseAndRequest';
+import { fetchRedeemablePolicies } from '../components/app/routes/queries/subsidies/policies';
+import { fetchSubscriptions } from '../components/app/routes/queries/subsidies/subscriptions';
+import { fetchCouponCodes } from '../components/app/routes/queries/subsidies/couponCodes';
+import { fetchEnterpriseOffers } from '../components/app/routes/queries/subsidies/enterpriseOffers';
+import { fetchEnterpriseCuration } from '../components/app/routes/queries/contentHighlights';
+
+// The keys that map to enterpriseQueryKeys that is used to define specific queries
+// and used to specifically invalidate queries
+const enterprise = createQueryKeyStore('enterprise', {
+  enterpriseCustomer: (enterpriseUuid) => ({
+    queryKey: [enterpriseUuid],
+    contextQueries: {
+      course: (courseKey) => ({
+        queryKey: [courseKey],
+        contextQueries: {
+          contentMetadata: {
+            queryKey: null,
+            queryFn: async ({ queryKey }) => fetchCourseMetadata(queryKey[2], queryKey[4]),
+          },
+          canRedeem: (availableCourseRunKeys) => ({
+            queryKey: [availableCourseRunKeys],
+            queryFn: async ({ queryKey }) => fetchCanRedeem(queryKey[2], availableCourseRunKeys),
+          }),
+        },
+      }),
+      user: {
+        queryKey: ['entitlements'],
+        queryFn: async () => fetchUserEntitlements(),
+      },
+      enrollments: {
+        queryKey: null,
+        queryFn: async ({ queryKey }) => fetchEnterpriseCourseEnrollments(queryKey[2]),
+      },
+      contentHighlights: {
+        queryKey: null,
+        contextQueries: {
+          configuration: {
+            queryKey: null,
+            queryFn: async ({ queryKey }) => fetchEnterpriseCuration(queryKey[2]),
+          },
+        },
+      },
+    },
+  }),
+  enterpriseLearner: (username, enterpriseSlug) => ({
+    queryKey: [username, enterpriseSlug],
+    queryFn: async () => fetchEnterpriseLearnerData(username, enterpriseSlug),
+  }),
+  userSubsidy: (enterpriseUuid) => ({
```

Comment:
Enterprise subsidies should be nested under the `enterpriseCustomer` query key as a contextual query, given the subsidies returned are specific to the enterprise customer UUID. 

Also, nit / no strong feelings, but tempted to rename this key as `subsidies` to be a bit more semantically correct given that not all subsidies-related queries here are specific to an individual learner (could be broadly applicable to the entire enterprise customer). By using `subsidies` vs. `userSubsidy`, it perhaps reduces any confusion about explicitly assigned subsidies (e.g., subscriptions, codes)  vs. auto-applied subsidies (e.g., auto-applied LC budgets, enterprise offers).